internal/datacoord/compaction_task_l0.go

```diff
@@ -45,116 +46,144 @@ type l0CompactionTask struct {
 	taskProto atomic.Value // *datapb.CompactionTask
 
 	allocator allocator.Allocator
-	sessions  session.DataNodeManager
 	meta      CompactionMeta
+
+	times *taskcommon.Times
 }
 
-func (t *l0CompactionTask) GetTaskProto() *datapb.CompactionTask {
-	task := t.taskProto.Load()
-	if task == nil {
-		return nil
-	}
-	return task.(*datapb.CompactionTask)
+func (t *l0CompactionTask) GetTaskID() int64 {
+	return t.GetTaskProto().GetPlanID()
 }
 
-func newL0CompactionTask(t *datapb.CompactionTask, allocator allocator.Allocator, meta CompactionMeta, session session.DataNodeManager) *l0CompactionTask {
-	task := &l0CompactionTask{
-		allocator: allocator,
-		meta:      meta,
-		sessions:  session,
-	}
-	task.taskProto.Store(t)
-	return task
+func (t *l0CompactionTask) GetTaskType() taskcommon.Type {
+	return taskcommon.Compaction
 }
 
-// Note: return True means exit this state machine.
-// ONLY return True for Completed, Failed
-func (t *l0CompactionTask) Process() bool {
-	switch t.GetTaskProto().GetState() {
-	case datapb.CompactionTaskState_pipelining:
-		return t.processPipelining()
-	case datapb.CompactionTaskState_executing:
-		return t.processExecuting()
-	case datapb.CompactionTaskState_meta_saved:
-		return t.processMetaSaved()
-	case datapb.CompactionTaskState_completed:
-		return t.processCompleted()
-	case datapb.CompactionTaskState_failed:
-		return t.processFailed()
-	}
-	return true
+func (t *l0CompactionTask) GetTaskState() taskcommon.State {
+	return taskcommon.FromCompactionState(t.GetTaskProto().GetState())
 }
 
-func (t *l0CompactionTask) processPipelining() bool {
-	if t.NeedReAssignNodeID() {
-		return false
-	}
+func (t *l0CompactionTask) GetTaskSlot() int64 {
+	batchSize := paramtable.Get().CommonCfg.BloomFilterApplyBatchSize.GetAsInt()
+	factor := paramtable.Get().DataCoordCfg.L0DeleteCompactionSlotUsage.GetAsInt64()
+	return factor * t.GetTaskProto().GetTotalRows() / int64(batchSize)
```

Comment:
Warning:  GetTaskSlot should never return 0.

Reply:
fixed